compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -113,9 +114,22 @@ object ProtoTypes {
      *  achieved by replacing expected type parameters with wildcards.
      */
     def constrainResult(meth: Symbol, mt: Type, pt: Type)(using Context): Boolean =
-      if (Inlines.isInlineable(meth) && meth.is(Transparent)) {
-        constrainResult(mt, wildApprox(pt))
-        true
+      if (Inlines.isInlineable(meth)) {
+        // Stricter behaviour in 3.4+: do not apply `wildApprox` to non-transparent inlines
+        if (Feature.sourceVersion.isAtLeast(SourceVersion.future)) {
```

Comment:
I guess there would be another follow up to add the `3.4` source version and enable it there?

Reply:
Yes, I’ve created #18174 to track that.